torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -776,6 +776,58 @@ class ShapePropagator {
       }
     };
 
+    // Requirements:
+    //   dims           : preserved
+    //   scalar type    : promoted to floating if integral/boolean, else preserved
+    //   device         : preserved
+    //   tensor inputs  : 1
+    //   tensor outputs : 1
+    // Additionally:
+    //   - First input should be the only tensor input
+    static const register_formula_for unary_floating_ufuncs{
+      {
+        "aten::acos(Tensor self) -> Tensor",
+        "aten::angle(Tensor self) -> Tensor",
+        "aten::tanh(Tensor self) -> Tensor",
+        "aten::asin(Tensor self) -> Tensor",
+        "aten::atan(Tensor self) -> Tensor",
+        "aten::ceil(Tensor self) -> Tensor",
+        "aten::cos(Tensor self) -> Tensor",
+        "aten::cosh(Tensor self) -> Tensor",
+        "aten::digamma(Tensor self) -> Tensor",
+        "aten::erf(Tensor self) -> Tensor",
+        "aten::erfc(Tensor self) -> Tensor",
+        "aten::erfinv(Tensor self) -> Tensor",
+        "aten::exp(Tensor self) -> Tensor",
+        "aten::expm1(Tensor self) -> Tensor",
+        "aten::log(Tensor self) -> Tensor",
+        "aten::log10(Tensor self) -> Tensor",
+        "aten::log1p(Tensor self) -> Tensor",
+        "aten::log2(Tensor self) -> Tensor",
+        "aten::floor(Tensor self) -> Tensor",
+        "aten::lgamma(Tensor self) -> Tensor",
+        "aten::rsqrt(Tensor self) -> Tensor",
+        "aten::sigmoid(Tensor self) -> Tensor",
+        "aten::sin(Tensor self) -> Tensor",
+        "aten::sinh(Tensor self) -> Tensor",
+        "aten::sqrt(Tensor self) -> Tensor",
+        "aten::tan(Tensor self) -> Tensor",
+        "aten::tanh(Tensor self) -> Tensor",
+        "aten::trunc(Tensor self) -> Tensor",
+      },
+      [](Node* node) -> type_vec_t {
+        if (auto type = node->input(0)->type()->cast<TensorType>()) {
+          auto ret = type->withDim(0);
```

Comment:
Do you need this withDim? It doesn't appear in the simple_unary_ops lambda.

Reply:
You are right. Thank you!